scalding-core/src/main/scala/com/twitter/scalding/reducer_estimation/Common.scala

```diff
@@ -23,6 +23,12 @@ object EstimatorConfig {
   /** Maximum number of history items to use for reducer estimation. */
   val maxHistoryKey = "scalding.reducer.estimator.max.history"
 
+  /** If we estimate more than this number of reducers, consider it a fatal error */
+  val maxEstimatedReducersKey = "scalding.reducer.estimator.max.estimated.reducers"
+
+  // TODO: what's a reasonable default? Int.maxValue? 5k? 100k?
+  val defaultMaxEstimatedReducers = 100 * 1000
```

Comment:
One reason I can think of - too many reducers equals too many files. 

Yeah, I don't know the historic reason for 999 reducer. I think Hive has the same too.